IMAGINGCLOUD-216

s-zubov opened the pull request.

s-zubov committed:
Add generated code

Add code generated by swagger-codegen from
Aspose.Imaging Cloud API Reference v3.0

s-zubov committed:
Remove unnecessary files

Remove generated test files, as they should
be created manually. Remove unnecessary files
created by codegen. Add .swagger-codegen-ignore to .gitignore

s-zubov committed:
Rename swagger_client project

Rename swagger_client project to asposeimagingcloud

s-zubov committed:
Add api_tester

s-zubov committed:
Add token requesting

s-zubov committed:
Fix SDK

s-zubov committed:
Fix test_bmp_api

s-zubov committed:
Fix content-type for request with image

s-zubov committed:
Fix receiving response with image

s-zubov committed:
Add test_post_request

s-zubov committed:
Add test_post_image_bmp

s-zubov committed:
Add test_crop_api

s-zubov committed:
Add test_emf_api

s-zubov committed:
Add test_examples

s-zubov committed:
Add test_frames_get_api

s-zubov committed:
Fix content-type for frames api

s-zubov committed:
Add test_frames_post_api

s-zubov committed:
Add test_gif_api

s-zubov committed:
Add test_jpeg2000_api

s-zubov committed:
Add test_jpg_api

s-zubov committed:
Add test_pds_api

s-zubov committed:
Add test_resize_api

s-zubov committed:
Add test_rotate_flip_api

s-zubov committed:
Add test_save_as_api

s-zubov committed:
Add test_tiff_api

s-zubov committed:
Add test_update_image_api

s-zubov committed:
Add test_web_p_api

s-zubov committed:
Add test_wmf_api

s-zubov committed:
Add storage_api_tester

s-zubov committed:
Add test_storage_api

s-zubov committed:
Fix file_version model

s-zubov committed:
Add test_file_api

s-zubov committed:
Add test_folder_api

s-zubov committed:
Add EXTENDED_TEST switch

s-zubov committed:
Modify api_tester methods' names

Naming methods like 'test_smth' causes test loader to add them to test
suite, so test_get_request and test_post_request renamed to
get_request_tester and post_request_tester

s-zubov committed:
Add test runner

s-zubov committed:
Update tests for using request models

s-zubov committed:
Update generated code

s-zubov committed:
Update generated code

s-zubov committed:
Add test image

s-zubov committed:
Update reading config in api_tester

s-zubov committed:
Fix api_tester

s-zubov committed:
Update generated code

s-zubov committed:
Update gitignore

s-zubov committed:
Remove .idea folder

s-zubov committed:
Update generated code

s-zubov committed:
Modify tests to run with Python2.7

s-zubov committed:
Modify tests setup and output

s-zubov committed:
Apply codestyle for tests

s-zubov committed:
Revert "Modify tests setup and output"

This reverts commit b7182f6f

s-zubov committed:
Modify tests setup and output

s-zubov committed:
Add AI API tester

s-zubov committed:
Add test_compare_images

s-zubov committed:
Fix exception handling in api

s-zubov committed:
Add AI tests

s-zubov committed:
Update generated code

s-zubov committed:
Fix file length checks

s-zubov committed:
Fix test package

s-zubov committed:
Fix image features model

s-zubov committed:
Modify test runner output

s-zubov committed:
Add test case name output

s-zubov committed:
Fix protocol for API

s-zubov committed:
Remove unused code from test setup

s-zubov committed:
Update generated code

Commit:
Update test method name output to console

s-zubov committed:
Fix base test class fields

s-zubov committed:
Update generated code

s-zubov committed:
Improve codestyle for tests

s-zubov committed:
Add copyright to tests

s-zubov committed:
Update generated code

s-zubov committed:
Fix copyright

MaxCrank committed:
Merge pull request #3 from aspose-imaging-cloud/PROBATIONIMAGING-509

PROBATIONIMAGING-509 - Prepare Aspose.Imaging Cloud Python SDK prototype

s-zubov committed:
Update SDK to v19.6

s-zubov committed:
Update tests to v19.6

s-zubov committed:
Add license

s-zubov committed:
Add readme stub

s-zubov committed:
Update requirements

s-zubov committed:
Update generated code

s-zubov committed:
Add setup.py

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Add EXAMPLES.md

s-zubov committed:
Add README.md

s-zubov committed:
Modify imaging_api constructor

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Fix test_examples

s-zubov committed:
Add xml output and test categories

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Revert "SDK regenerated by CI server"

This reverts commit 58da10d9

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Fix typo in delete_image_search

s-zubov committed:
Fix test_search_context

s-zubov committed:
Fix AI tests folder

s-zubov committed:
Bump version to 19.7.0

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

MaxCrank commented:
@s-zubov Please, set me as a reviewer and yourself as an assignee to all pull requests (there's a common misconception about these roles in GitHub which had also been mine for some time, too). Thanks.

MaxCrank commented on README.md:
```diff
@@ -0,0 +1,96 @@
+# Aspose.Imaging Cloud Python SDK 
+[Aspose.Imaging Cloud](https://products.aspose.cloud/imaging/cloud) is a true REST API that enables you to perform a wide range of image processing operations including creation, manipulation and conversion in the cloud, with zero initial costs. Our Cloud SDKs are wrappers around REST API in various programming languages, allowing you to process images in language of your choice quickly and easil ...
```
@s-zubov Please, change the link to https://products.aspose.cloud/imaging - this is a new requirement from the Cloud team which I agree with. Thanks.

s-zubov replied:
Thank you, I've updated the link.

MaxCrank commented on README.md:
```diff
@@ -0,0 +1,96 @@
+# Resources
++ [**Web API reference**](https://apireference.aspose.cloud/imaging/)
++ [**Website**](https://www.aspose.cloud)
++ [**Product Home**](https://products.aspose.cloud/imaging/cloud)
```
@s-zubov Please, change the link to https://products.aspose.cloud/imaging - this is a new requirement from the Cloud team which I agree with. Thanks.

s-zubov replied:
Thank you, I've updated the link.

MaxCrank commented on README.md:
```diff
@@ -0,0 +1,96 @@
+* Check if certain file, folder or storage exists
+
+# Usage
+Please, add the following [NuGet package](https://www.nuget.org/packages/Aspose.Imaging-Cloud/) to your project.
```
@s-zubov Please, provide the appropriate instructions and future link for package manager instead of NuGet.

s-zubov replied:
Thank you, I've added a link to the PyPi website as the package is not published yet.

MaxCrank replied:
@s-zubov would you mind to add an actual to-be link (as if it was released)? Thanks.

MaxCrank replied:
@s-zubov Also, please provide an example on how to include this package in the project, considering other SDKs as an example. The reason why it's not done in .NET SDK is that NuGet package page includes this info right away, while other package managers (including PyPI) don't.

s-zubov replied:
Thank you, I've provided detailed information about installing package and using it.

MaxCrank replied:
@s-zubov Please, check your latest commit - there's probably something wrong with _.gitignore_ in this repo or with _.swagger-codegen-ignore_ in Python codegen templates. Only README had to be changed.

MaxCrank replied:
@s-zubov Seems that I've missed a major issue with package name - it should be **_asposeimagingcloud_** instead of **_aspose-imaging-cloud-sdk_**. Please, change it and update all dependencies (imports, README, links etc). Thanks.

MaxCrank replied:
@s-zubov sorry for the confusion caused by my recent comment. We had a lot of discussions about naming conventions, and it seems that the right package name is simply _**aspose-imaging-cloud**_. Thanks.

s-zubov replied:
@MaxCrank platform-specific guidelines are not taken into consideration? According to PEP, package name should be **asposeimagingcloud**, not **aspose_imaging_cloud** or **aspose-imaging-cloud**. Of course, PEP is just a recommendation.

> Python packages should also have short, all-lowercase names, although the use of underscores is discouraged.
https://www.python.org/dev/peps/pep-0008/#package-and-module-names

s-zubov replied:
I've updated package name to **aspose-imaging-cloud** according to company standards.

MaxCrank commented on README.md:
```diff
@@ -0,0 +1,96 @@
+Please, add the following [NuGet package](https://www.nuget.org/packages/Aspose.Imaging-Cloud/) to your project.
+
+# Examples
+This SDK will be published to pip soon.
```
@s-zubov Please, add the link to examples as in other SDKs.

s-zubov replied:
Thank you, I've added a link to the examples.

MaxCrank commented on README.md:
```diff
@@ -0,0 +1,96 @@
+# Aspose.Imaging Cloud Python SDK 
```
@s-zubov Please, enrich the README with info about async requests in a manner which is used in [PHP SDK](https://github.com/aspose-imaging-cloud/aspose-imaging-cloud-php/blob/master/README.md#asynchronous-api). Thanks.

s-zubov replied:
Thank you, I've provided information about async requests at the readme.

MaxCrank commented on asposeimagingcloud/rest.py:
```diff
@@ -0,0 +1,341 @@
+                            body=body)
+
+
+class ApiException(Exception):
```
@s-zubov Please, refactor the _ApiException_ class and exception handling example, looking at other SDKs which include common error format data, and using the [existing model](https://github.com/aspose-imaging-cloud/aspose-imaging-cloud-python/blob/IMAGINGCLOUD-216/asposeimagingcloud/models/error.py) for this. Please, make the SDK fail and ensure that the appropriate exception is raised and the common error response data is properly deserialized. Thanks.

s-zubov replied:
Thank you, I've refactored exception handling. It is a bit tricky, in Python SDK no third-party library is used to deserialize responses from the server. I've rearranged the code that carries deserialization so it is accessible from exception processing function. 

This is an example of an error occurred not on the cloud servers (so, no internal data provided):
> ('502 Error connecting to the API https://api-qa.aspose.cloud/v3.0/imaging/storage/folder/ImagingIntegrationTestData; Message: Bad Gateway', 502, None)

I'll provide an example with captured server error later.

s-zubov replied:
This is an example of an error occurred on the server:

> ('404 Error connecting to the API https://api-qa.aspose.cloud/v3.0/imaging/d23d2d/bmp; Message: Not Found', 404, {'code': 'errorItemNotFound',
 'description': 'Operation Failed. Item Not Found.',
 'inner_error': {'_date': datetime.datetime(2019, 7, 19, 7, 23, 2, 991837, tzinfo=tzutc()),
                 'request_id': None},
 'message': "Item doesn't exist at path ImagingCloudTestPython_szubov/d23d2d."})

MaxCrank commented on setup.py:
```diff
@@ -0,0 +1,111 @@
+        "jpx",
+        "jpm",
+        "j2k",
+        "odg",
```
@s-zubov Please, add the _otg_ tag as we will add it in v19.7 as well. Thanks.

s-zubov replied:
Thank you, I've added the otg tag.

s-zubov committed:
Update README

Fix links to Aspose.Imaging Cloud.
Fix Usage and Examples.
Add information about async requests.

s-zubov committed:
Add otg tag to package info

s-zubov committed:
Modify ApiException

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov commented:
> @s-zubov Please, confirm if it's a valid commit caused by SDK generation; otherwise, please fix the codegen templates. Thanks.

I've updated codegen templates, the previous commit was incomplete.

Commit:
Add usage details to README

aspose-imaging-cloud-deployer committed:
SDK regenerated by CI server

s-zubov committed:
Modify package name to aspose-imaging-cloud
